components/ButtonComponent.tsx

```diff
@@ -5,16 +5,23 @@ interface Props {
   bgColor: string;
   title: string;
   buttonCallback: () => void;
+  buttonDisabled: boolean;
```

Comment:
you could make it optional:
```
buttonDisabled?: boolean;
```
as then you don't have to pass it down (and prop not passed means `undefined`, which is falsy, so it works the same as passing `false` but is a bit cleaner IMO)